### PR TITLE
restructure kepler model server with get started page

### PR DESCRIPTION
**Pull request by sunya-ch**
This PR restructures the content related to Kepler Model Server. 
Changes includes
- now estimator is merged to model server project.
- prometheus name (`kepler_container_other_joules_total`) change in kepler according to PR https://github.com/sustainable-computing-io/kepler/pull/871.
- add `get started` page for kepler model server 

might relate to https://github.com/sustainable-computing-io/kepler-doc/issues/67

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

### CODE REVIEW

File: docs/kepler_model_server/pipeline.md [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
 [rapl](../design/kepler-energy-sources.md#rapl---running-average-power-limit)|package, core, uncore, dram
 [acpi](../design/kepler-energy-sources.md#using-kernel-driver-xgene-hwmon)|platform
+||
 
-## Model Features
-The features are groupped by the metric sources.
+## Idle power/Control plane power
 
-### Feature group
+`isolate` is a mechanism to separate the power portion that is consumed on the node in an idle state or the power portion that is consumed by operating systems and control plane processes. These portions of power is more than zero even if the metric utilization of workload is zero. We called the models those are trained after isolating these power portions as `DynPower` models. At the same time, the models those are trained without isolation are called `AbsPower` models. `DynPower` model is used to estimate `container power` and `process power` while `AbsPower` model is used to estimate `node power`.
```
Comment (marceloamaral) [inline docs/kepler_model_server/pipeline.md:38]: > [...] the power portion that is consumed by operating systems and control plane processes. These portions of power is more than zero even if the metric utilization of workload is zero.

What do you mean about that?

The idle power is higher than 0 if the utilization of workload is zero, but the dynamic power should be 0 if the utilization of workload is zero!

Although we might not have the OS CPU utilization via cgroup or kubelet metrics, the eBPF metrics will provide the OS and control-plane CPU utilization which must match the resource utilization with the power utilization.

Additionally, we should maybe have a better distinction between isolating idle power and isolating other workload dynamic power (i.e., removing the control plane).

**Review by rootfs — APPROVED** [review]

File: docs/kepler_model_server/pipeline.md [hunk 2]
```diff
@@ -1,25 +1,54 @@
...
+
+There are two common available `isolators`: *ProfileIsolator* and *MinIdleIsolator*. 
+
+*ProfileIsolator* relies on profiled background powers (profiles) and removes resource usages by system processes from the training while *MinIdleIsolator* assumes minimum power as an idle power and includes resource usages by system processes in the training. 
+
+The pipeline with *ProfileIsolator* will be applied first if the profile that matches the training `node_type` is available. Otherwise, the other pipeline will be applied.
```
Comment (marceloamaral) [inline docs/kepler_model_server/pipeline.md:44]: > The pipeline with *ProfileIsolator* will be applied first if the profile that matches the training `node_type` is available.

What does it mean? What is the `node_type`? When it does not match?

> Otherwise, the other pipeline will be applied.

Can you explain it better?

> *ProfileIsolator* relies on profiled background powers (profiles) and removes resource usages by system processes from the training

If we don't know the power consumption of the system processes before hand, how can we remove the resource usages of the system processes?

File: docs/kepler_model_server/node_profile.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Node Profile
+
+To form a group of machines (nodes), the idea is to run a benchmark suite towards a bunch of machines, collect performance reported by the benchmarks, and apply clustering algorithm such as *kmeans*.
+
+For each group of [node type](./pipeline.md#node-spec), we make a profile composing of background power when the resource usage is almost constant without user workload, minimum, maximum power for each power components (e.g., core, uncore, dram, package, platform), and normalization scaler (i.e., MinMaxScaler), standardization scaler (i.e., StandardScaler) for each [feature group](./pipeline.md#available-metrics).
```
[the remark this answers is not in the document]
Comment (marceloamaral) [reply docs/kepler_model_server/node_profile.md:5]: Can you provide more details about the normalization scaler and standardization scaler? Maybe include the formulas and why we use them here.

Also, how they are they selected?

[the remark this answers is not in the document]
Comment (marceloamaral) [reply docs/kepler_model_server/node_profile.md:5]: The Kepler code has the Liner Regression normalization as:
```
normalizedX := (vals[index] - coeff.Mean) / math.Sqrt(coeff.Variance)
```
However, as I mentioned before the Mean can be a very high value and then the low utilization will return a negative value, which means a negative power consumption. Therefore, we must change this normalization.

**Comment by marceloamaral** [comment]
@sunya-ch although @rootfs has merged this PR, could you please reply my comments?

File: docs/kepler_model_server/pipeline.md (hunk 1 above)
Comment (sunya-ch) [reply docs/kepler_model_server/pipeline.md:38]: I would like to say about the part that we isolated from the AbsPower (AbsPower - DynPower).
Yes, I agree that we should elaborate more about the isolation. Because in the control plane case, it mean we also remove utilization from the control plane process too (done by trainIsolator).  Please feel free to modify the doc.

File: docs/kepler_model_server/pipeline.md (hunk 2 above)
Comment (sunya-ch) [reply docs/kepler_model_server/pipeline.md:44]: It took the Kepler metric in idle state. The containers that run in idle state is considered as system processes/control plane.

File: docs/kepler_model_server/node_profile.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sunya-ch) [reply docs/kepler_model_server/node_profile.md:5]: It uses the standard scaler from [scikit learn](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html) library.  Specifically, it uses [SGDRegressor](https://github.com/sustainable-computing-io/kepler-model-server/blob/main/src/train/trainer/SGDRegressorTrainer/main.py) to generate the weight.

[the remark this answers is not in the document]
Comment (marceloamaral) [reply docs/kepler_model_server/node_profile.md:5]: > class sklearn.preprocessing.StandardScaler(*, copy=True, with_mean=True, with_std=True)
> Standardize features by removing the mean and scaling to unit variance.
> 
> The standard score of a sample x is calculated as:
> 
> z = (x - u) / s
> 
> where u is the mean of the training samples or zero if with_mean=False, and s is the standard deviation of the training samples or one if with_std=False.

@sunya-ch So, we can set `with_mean=False` to avoid the problem that I mentined.

[the remark this answers is not in the document]
Comment (marceloamaral) [reply docs/kepler_model_server/node_profile.md:5]: MinMaxScaler looks to see the right approach.
